test/optimizers/test_optimizers.py

```diff
@@ -141,6 +146,72 @@ def test_sr3_bad_parameters(optimizer, params):
         optimizer(**params)
 
 
+@pytest.mark.parametrize(
+    "params",
+    [
+        dict(eta=-1),
+        dict(tol=0),
+        dict(tol_m=0),
+        dict(eps_solver=0),
+        dict(alpha_m=-1),
+        dict(alpha_A=-1),
+        dict(gamma=1),
+        dict(evolve_w=False, relax_optim=False),
+        dict(thresholder="l0"),
+        dict(threshold=-1),
+        dict(max_iter=0),
+        dict(eta=10, alpha_m=20),
+        dict(eta=10, alpha_A=20),
+    ],
+)
+def test_trapping_bad_parameters(params):
+    with pytest.raises(ValueError):
+        TrappingSR3(**params)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [dict(PL=np.random.rand(3, 3, 3, 9)), dict(PQ=np.random.rand(3, 3, 3, 3, 9))],
+)
+def test_trapping_bad_tensors(params):
+    x = np.random.standard_normal((10, 9))
+    x_dot = np.random.standard_normal((10, 3))
+    with pytest.raises(ValueError):
+        model = TrappingSR3(**params)
+        model.fit(x, x_dot)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [dict(PL=np.ones((3, 3, 3, 9)), PQ=np.ones((3, 3, 3, 3, 9)))],
+)
+def test_trapping_quadratic_library(params):
```

Comment:
Should the name be `test_trapping_cubic_library`?

Reply:
Either way. The point here is that if PL and PQ are passed, then the library needs to be quadratic